### ISSUE TITLE
[Design]: Design User Experience for Long Submission Times for the 10-10CG 

### ISSUE THREAD

**Issue by jonathanjnelson** [department-of-veterans-affairs/va.gov-team#20320]
## Description 

_Design and document solutions for long submission times for the 10-10CG _ 

## Acceptance Criteria

- [ ] _Design solution_
- [ ] _Mockup UI_
- [ ] _Sync with content (if needed)_
- [ ] _Approval from product_
- [ ] _Handoff to FE_

**Comment by jonathanjnelson** [comment]
[For awarness](https://app.zenhub.com/workspaces/vsa---caregiver-5fff0cfd1462b6000e320fc7/issues/department-of-veterans-affairs/va.gov-team/18736)

**Comment by jonathanjnelson** [comment]
Hi @peggygannon 

The 10-10CG can take up to 2 mins to process on the backend. Therefore, we are looking to

-  add progressive messaging to the loading indicator on submission
-  after a designated time (yet to be determined) add an alert 
-  Add a note on the review page setting expectations for the user 


![Screen Shot 2021-02-24 at 11.55.26 AM.png](https://images.zenhubusercontent.com/5df3ac60534bc19675f4ca43/27205ff5-1a5b-4dd5-8b85-cde7aa5a7d96)

My understanding is the standard content is: 

"Please wait while we process your application."

And after 30 seconds: 

"We're sorry this is taking longer than expected. Thank you for your patience."

Does that sound correct? 

Can we customize the messaging to something like: 

"We're sorry. This may take 15 seconds up to 2 minutes to process. Please do not refresh the browser."

**Comment by jonathanjnelson** [comment]
An alert might be needed due to the long wait times

![image (6).png](https://images.zenhubusercontent.com/5df3ac60534bc19675f4ca43/523afa51-feb5-4eeb-9c82-dbd23967e905)

**Comment by peggygannon** [comment]
Hi, @jonathanjnelson I have a few question about this: 

I want to make sure I understand when users will see this message, is it after they hit submit button,  but before they see the confirmation screen?

**Comment by jonathanjnelson** [comment]
@peggygannon 

Yes, that is correct

**Comment by peggygannon** [comment]
@jonathanjnelson  Here's content recommendation for after 30 second spinner message. 

- Standard process time:  Please wait while we process your application.

- And after 30 seconds: We’re processing your application. This may take up to 2 minutes. Please don’t refresh your browser. 

I don't think we need an alert in addition to this updated spinner message^^.  But I do recommend one if processing goes past the 2 minute time frame.  Is that content that's needed for this ticket too? 


- re: *Add a note on the review page setting expectations for the user*. Can you share a screenshot of this page?

**Comment by jonathanjnelson** [comment]
@peggygannon I agree with you on the alert. If possible, it would be great to help with the content of the alert in case we want to show after that given processing time. 

Here is a screenshot of the current review page  **Add a note on the review page setting expectations for the user.** 

![Screen Shot 2021-02-24 at 3.31.32 PM.png](https://images.zenhubusercontent.com/5df3ac60534bc19675f4ca43/595118f8-de30-4db5-ab7c-daf8037fadfb)

**Comment by peggygannon** [comment]
Hi, @jonathanjnelson Yes, I can recommend content for that alert! 

What is the path to resolution if the caregiver application takes longer than {designated time} to process?  Does the user make a call? do they refresh page and try again? do they have to resubmit app?

**Comment by jonathanjnelson** [comment]
@peggygannon Regarding the alert, we need to direct the user to make a phone call

**Comment by peggygannon** [comment]
thanks, @jonathanjnelson Please see content recommendations below. let me know if you have any questions. 

Question: for the alert where should we direct users to call? I put the help desk number in the  alert below, but should we use the Caregiver VA number? pls advise. 

**We’re having trouble processing your application**
We're sorry. Something went wrong when we tried to process your application. For help submitting your application, please call the VA.gov help desk at 800-698-2411 (TTY: 711). We’re here 24/7.


-----
Review page note:

**Note:** It can take several minutes for your application to process after you submit it. Please don’t refresh your browser during that time.

**Comment by jonathanjnelson** [comment]
Thank you @peggygannon. Yes, we should direct the users to HEC. They are trained to answer questions on the program/the 10-10CG. I plugged in the appropriate number/hours: 

We’re having trouble processing your application
We're sorry. Something went wrong when we tried to process your application. For help submitting your application, please call the VA.gov help desk at 855-260-3274 (TTY: 711). We’re here Monday through Friday, 8:00 a.m. to 10:00 p.m. ET, and Saturday, 8:00 a.m. to 5:00 p.m. ET.

**Comment by jonathanjnelson** [comment]
**Recommendations:**

**Loading indicator content:**

- Please wait while we process your application.

- And after 30 seconds (This can be adjusted if we want to show earlier): We’re processing your application. This may take up to 2 minutes. Please don’t refresh your browser.

**My recommendation is to not show the alert. However, if we decide that is best I don't think we need to change the content.**

![Screen Shot 2021-02-25 at 9.41.31 AM.png](https://images.zenhubusercontent.com/5df3ac60534bc19675f4ca43/fab04baf-f9eb-4d68-93d1-8a23a66b0612)


**Note on review and submit page:**
Note: It can take several minutes for your application to process after you submit it. Please don’t refresh your browser during that time.

**Error (when we want to direct user to HEC):**

We’re having trouble processing your application
We're sorry. Something went wrong when we tried to process your application. For help submitting your application, please call the VA.gov help desk at 855-260-3274 (TTY: 711). We’re here Monday through Friday, 8:00 a.m. to 10:00 p.m. ET, and Saturday, 8:00 a.m. to 5:00 p.m. ET.

**Comment by mlighdc** [comment]
@jonathanjnelson - if it is over 3 minutes, are you showing the call HEC message?

**Comment by jonathanjnelson** [comment]
Please see https://preview.uxpin.com/669525e992a5edd248b84db96767fc607630185b#/pages/137132433/simulate/no-panels?mode=i

**Comment by jonathanjnelson** [comment]
**Add Note to Review and Submit page**

![Screen Shot 2021-03-02 at 9.47.23 AM.png](https://images.zenhubusercontent.com/5df3ac60534bc19675f4ca43/3f4aeacd-82df-4e44-b90b-5c11c830dc27)

**Comment by jonathanjnelson** [comment]
**Loading Indicator with Message**

![Screen Shot 2021-03-02 at 9.48.59 AM.png](https://images.zenhubusercontent.com/5df3ac60534bc19675f4ca43/89d7d733-200d-4831-99f3-a29abcf0de94)

**Comment by jonathanjnelson** [comment]
**Loading Indicator Message progresses after 30 seconds:** 

![Screen Shot 2021-03-02 at 9.51.52 AM.png](https://images.zenhubusercontent.com/5df3ac60534bc19675f4ca43/f323cf2e-43fe-4da5-aeb8-7124b14b5c3f)

**Comment by jonathanjnelson** [comment]
Please see https://app.zenhub.com/workspaces/vsa---caregiver-5fff0cfd1462b6000e320fc7/issues/department-of-veterans-affairs/va.gov-team/20508

In order to move forward with the error state, the team needs to determine: 

What information should a Veteran see?
Can we address the long CARMA processing times?
Can we try to resend the PDF?
